src/v/cloud_topics/level_one/metastore/state_update.cc

```diff
@@ -25,12 +25,26 @@ struct extent_range {
 std::optional<extent_range> get_range(
   const partition_state::extent_set_t& extents,
   kafka::offset base,
-  kafka::offset last) {
+  kafka::offset last,
+  kafka::offset log_start_offset) {
     auto base_it = std::ranges::lower_bound(
       extents, base, std::less<>{}, &extent::base_offset);
-    if (base_it == extents.end() || base_it->base_offset != base) {
+    if (base_it == extents.end()) {
         return std::nullopt;
     }
+    if (base_it->base_offset != base) {
+        if (base < log_start_offset) {
+            // We are considering replacing an extent below the log start
+            // offset. We allow for misalignment between the start offset and
+            // the first extent in the partition, since we need to support
+            // partial truncations- this also means we don't need replacement
+            // extents to be aligned to existing extents below the start offset.
+            // Fallthrough here instead of returning `std::nullopt`- we will
+            // still need to validate the alignment of the range's last offset.
+        } else {
+            return std::nullopt;
+        }
+    }
```

Comment:
I think it'll be easier to reason about if we don't bother with the lower_bound when base < log_start_offset and instead return the start of the extents in that case, something like:

```
auto base_it = extents.begin();
if (base >= log_start_offset) {
    base_it = std::ranges::lower_bound(
      extents, base, std::less<>{}, &extent::base_offset);
    if (base_it == extents.end() || base_it->base_offset != base) {
        return std::nullopt;
    }
}
```